premailer/cache.py

```diff
@@ -1,15 +1,43 @@
 import functools
 import os
+import threading
 
-from cachetools import cached, LRUCache
+import cachetools
 
 
-cache = LRUCache(maxsize=int(os.environ.get("PREMAILER_CACHE_MAXSIZE", 128)))
+# Available cache options.
+CACHE_IMPLEMENTATIONS = {
+    "LFU": cachetools.LFUCache,
+    "LRU": cachetools.LRUCache,
+    "RR": cachetools.RRCache,
+    "TTL": cachetools.TTLCache,
+}
+
+# Time to live (seconds) for entries in TTL cache. Defaults to 1 hour.
+TTL_CACHE_TIMEOUT = 1 * 60 * 60
+
+# Lock to prevent multiple threads from accessing the cache at same time.
+CACHE_ACCESS_LOCK = threading.RLock()
```

Comment:
This should be lowercase since it's not really a human chosen constant and it's never really configurable. Note that it's called `cache` on line 35 (not `CACHE`).